cognite/neat/_session/_state_machine/_states.py

```diff
@@ -0,0 +1,148 @@
+        elif event == "read_physical" or event == "convert_physical":
+            return ConceptualPhysicalState()
+        elif event == "transform_conceptual":
+            return ConceptualState()
+        elif event == "write_conceptual":
+            return ConceptualState()
+
+        return ForbiddenState(self)
+
+
+class PhysicalState(State):
+    """
+    State with physical model loaded.
+    """
+
+    def on_event(self, event: str) -> State:
+        if event == "read_physical":
+            return PhysicalState()
+        elif event == "transform_physical":
+            return PhysicalState()
+        elif event == "write_physical":
+            return PhysicalState()
+        elif event == "convert_physical" or event == "read_conceptual":
+            return ConceptualPhysicalState()
+
+        return ForbiddenState(self)
+
+
+class InstancesConceptualState(State):
+    """
+    State with both instances and conceptual model loaded.
+    """
+
+    def on_event(self, event: str) -> State:
+        if event == "read_conceptual":
+            return InstancesConceptualState()
+        elif event == "transform_conceptual":
+            return InstancesConceptualState()
+        elif event in ["write_instances", "write_conceptual"]:
+            return InstancesConceptualState()
+        elif event in ["read_physical", "convert_physical"]:
+            return InstancesConceptualPhysicalState()
+
+        return ForbiddenState(self)
+
+
+class ConceptualPhysicalState(State):
+    """
+    State with both conceptual and physical models loaded.
+    """
+
+    def on_event(self, event: str) -> State:
+        if event == "read_physical":
+            return ConceptualPhysicalState()
+        elif event == "transform_physical":
+            return ConceptualPhysicalState()
+        elif event in ["write_conceptual", "write_physical"]:
+            return ConceptualPhysicalState()
+
+        return ForbiddenState(self)
+
+
+class InstancesConceptualPhysicalState(State):
+    """
+    State with instances, conceptual, and physical models loaded.
+    """
+
+    def on_event(self, event: str) -> State:
+        if event == "read_physical":
+            return InstancesConceptualPhysicalState()
+        elif event == "transform_physical":
+            return InstancesConceptualPhysicalState()
+        elif event in ["write_instances", "write_conceptual", "write_physical"]:
+            return InstancesConceptualPhysicalState()
+
+        return ForbiddenState(self)
+
+
+class ForbiddenState(State):
+    """
+    State representing forbidden transitions - returns to previous state.
+    """
+
+    def __init__(self, previous_state: State):
+        self.previous_state = previous_state
+        print(f"Forbidden action attempted. Returning to previous state: {previous_state}")
+
+    def on_event(self, event: str) -> State:
+        # All events from forbidden state return to previous state
```

Comment:
looks confusing and contradicting NeatSession flow, there calling on_even on Forbidden state is illegal, instead transition doesn't happen / doesn't get committed, is it expected that forbidden/error state can be entered and exited?